cmake/toolchain/zephyr.cmake

```diff
@@ -27,7 +28,7 @@ elseif("${ARCH}" STREQUAL "xtensa")
   set(TOOLCHAIN_INCLUDES
     ${SYSROOT_DIR}/include/arch/include
     ${SYSROOT_DIR}/include
-    )
```

Comment:
It would be helpful from a review perspective, and from a PR maintainability perspective, if the patches that are desirable (even without applying the toolchain abstraction patches) could be upstreamed before the PR is ready for review.

Reply:
The PR has now been split across multiple commits. That takes out much of the noise from things like whitespace changes.
I think it is ready for review.